src/generators/cplusplus/CplusplusConstrainer.ts

```diff
@@ -0,0 +1,71 @@
+import { TypeMapping } from '../../helpers';
+import {
+  defaultEnumKeyConstraints,
+  defaultEnumValueConstraints
+} from './constrainer/EnumConstrainer';
+import { defaultModelNameConstraints } from './constrainer/ModelNameConstrainer';
+import { defaultPropertyKeyConstraints } from './constrainer/PropertyKeyConstrainer';
+import { CplusplusDependencyManager } from './CplusplusDependencyManager';
+import { CplusplusOptions } from './CplusplusGenerator';
+
+export const CplusplusDefaultTypeMapping: TypeMapping<
+  CplusplusOptions,
+  CplusplusDependencyManager
+> = {
+  Object({ constrainedModel }): string {
+    //Returning name here because all object models have been split out
+    return `${constrainedModel.name}*`;
+  },
+  Reference({ constrainedModel }): string {
+    return `${constrainedModel.name}*`;
+  },
+  Any(): string {
+    return 'void*';
+  },
+  Float(): string {
+    return 'double';
+  },
+  Integer(): string {
+    return 'int';
+  },
+  String({ dependencyManager }): string {
+    dependencyManager.addDependency('#include <string>');
+    return 'std::string';
+  },
+  Boolean(): string {
+    return 'bool';
+  },
+  Tuple({ constrainedModel, dependencyManager }): string {
+    const types = constrainedModel.tuple.map((model) => {
+      return model.value.type;
+    });
+    dependencyManager.addDependency('#include <tuple>');
+    return `std::tuple<${types.join(', ')}>`;
+  },
+  Array({ constrainedModel, dependencyManager }): string {
+    dependencyManager.addDependency('#include <vector>');
+    return `std::vector<${constrainedModel.valueModel.type}>`;
+  },
+  Enum({ constrainedModel }): string {
+    //Returning name here because all enum models have been split out
+    return `${constrainedModel.name}*`;
```

Comment:
It doesn't make sense to use indirection here, store the enum by value instead. The `*` should be removed.